### PR TITLE
Bug 1912944: Set proxy env in vsphere-problem-detector, manila and oVirt deployment containers

### PR DESCRIPTION
**Pull request by bertinatto**
CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/hold
this is WIP

File: pkg/operator/configobservation/util/util.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// InjectObservedProxyInDeploymentContainers takes an observed proxy config and returns a patched Deployment with proxy env vars set.
+func InjectObservedProxyInDeploymentContainers(deployment *appsv1.Deployment, opSpec *operatorapi.OperatorSpec, observedConfig []byte, fields ...string) (*appsv1.Deployment, error) {
```
Comment (jsafrane) [inline pkg/operator/configobservation/util/util.go:11]: `observedConfig` and `fields` are not used anywhere

Comment (bertinatto) [reply pkg/operator/configobservation/util/util.go]: removed

**Comment by bertinatto** [comment]
/hold cancel
openshift/library-go#976 has been merged.
@openshift/storage PTAL

**Comment by jsafrane** [comment]
@bertinatto can you please update ovirt CSI operatot deployment too? I noticed it connects to oVirt API to generate the right storage classes: https://github.com/openshift/ovirt-csi-driver-operator/blob/1fda62f9be21bce86f2b0def579e81a4426ea4a0/pkg/operator/storageclass_controller.go#L85

**Comment by bertinatto** [comment]
> @bertinatto can you please update ovirt CSI operatot deployment too? I noticed it connects to oVirt API to generate the right storage classes: https://github.com/openshift/ovirt-csi-driver-operator/blob/1fda62f9be21bce86f2b0def579e81a4426ea4a0/pkg/operator/storageclass_controller.go#L85

@jsafrane upated.

**Comment by bertinatto** [comment]
level=error msg=Error: googleapi: Error 504: , gatewayTimeout 

/test e2e-gcp

**Pull request merged**